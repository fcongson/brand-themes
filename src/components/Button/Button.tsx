import * as React from 'react'
import styled from 'styled-components'

const PrimaryButton = styled.button`
  ${(props) => props?.theme?.button?.primary}
`

const SecondaryButton = styled.button`
  ${(props) => props?.theme?.button?.secondary}
`

const Button: React.FC<{ type?: 'primary' | 'secondary' }> = ({ type, children }) => {
  switch (type) {
    case 'primary':
      return <PrimaryButton>{children}</PrimaryButton>
    case 'secondary':
      return <SecondaryButton>{children}</SecondaryButton>
    default:
      return <button>{children}</button>
  }
}

export default Button
