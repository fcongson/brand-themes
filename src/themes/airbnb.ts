import WebFont from 'webfontloader'

WebFont.load({
  google: {
    families: ['Montserrat:700'],
  },
})

const color = {
  primary: '#008489',
  secondary: '#ffffff',
  white: '#ffffff',
}

const button = {
  primary: `
  background-color: ${color.primary};
  height: 44px;
  width: 183px;
  border-radius: 8px;
  border: 1px solid ${color.primary};
  color: ${color.white};
  font-family: Montserrat;
  font-size: 12px;
  font-weight: 700;
  cursor: pointer;
  `,
  secondary: `
  background-color: ${color.secondary};
  height: 44px;
  width: 183px;
  border-radius: 8px;
  border: 1px solid ${color.primary};
  color: ${color.primary};
  font-family: Montserrat;
  font-size: 12px;
  font-weight: 700;
  cursor: pointer;
  `,
}

const theme = {
  button,
}

export default theme
