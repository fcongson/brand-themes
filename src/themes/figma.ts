import WebFont from 'webfontloader'

WebFont.load({
  google: {
    families: ['Lato:700'],
  },
})

const color = {
  primary: '#000000',
  secondary: '#ffffff',
  white: '#ffffff',
  black: '#000000',
}

const button = {
  primary: `
  background-color: ${color.primary};
  height: 42px;
  width: 302px;
  border-radius: 8px;
  border: 3px solid ${color.primary};
  color: ${color.white};
  font-family: Lato;
  font-size: 16px;
  font-weight: 700;
  text-transform: none;
  cursor: pointer;
  &:hover {
    transform: translateY(-2px);
  }
  &:active {
    transform: translateY(1px);
  }
  transition: transform 200ms,background 200ms;
  `,
  secondary: `
  background-color: ${color.secondary};
  height: 42px;
  width: 302px;
  border-radius: 8px;
  border: 3px solid ${color.primary};
  color: ${color.black};
  font-family: Lato;
  font-size: 16px;
  font-weight: 700;
  text-transform: none;
  cursor: pointer;
  &:hover {
    transform: translateY(-2px);
  }
  &:active {
    transform: translateY(1px);
  }
  transition: transform 200ms,background 200ms;
  `,
}

const theme = {
  button,
}

export default theme
