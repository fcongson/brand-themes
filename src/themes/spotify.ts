import WebFont from 'webfontloader'

WebFont.load({
  google: {
    families: ['Montserrat:700'],
  },
})

const color = {
  primary: '#1db954',
  secondary: 'transparent',
  white: '#ffffff',
  black: '#000000',
}

const button = {
  primary: `
  background-color: ${color.primary};
  height: 48px;
  width: 214px;
  border-radius: 500px;
  border: none;
  color: ${color.white};
  font-family: Montserrat;
  font-size: 14px;
  font-weight: 700;
  text-transform: uppercase;
  cursor: pointer;
  &:hover {
    transform: scale(1.04);
    background-color: #1ed760;
  }
  &:active {
    transform: scale(1);
    background-color: #14833b;
  }
  transition: all 33ms;
  `,
  secondary: `
  background-color: ${color.secondary};
  height: 48px;
  width: 214px;
  border-radius: 500px;
  border: 2px solid ${color.black};
  color: ${color.black};
  font-family: Montserrat;
  font-size: 14px;
  font-weight: 700;
  text-transform: uppercase;
  cursor: pointer;
  &:hover {
    transform: scale(1.04);
  }
  &:active {
    transform: scale(1);
  }
  transition: all 33ms;
  `,
}

const theme = {
  button,
}

export default theme
