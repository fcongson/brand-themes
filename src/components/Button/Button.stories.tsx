import * as React from 'react'
import { ThemeProvider } from 'styled-components'
import airbnb from '../../themes/airbnb'
import theme from '../../themes/default'
import figma from '../../themes/figma'
import spotify from '../../themes/spotify'
import Button from './Button'

export default {
  title: 'Button',
  component: Button,
}

export const Default = () => {
  return <Button>Default</Button>
}

export const DefaultPrimary = () => {
  return (
    <ThemeProvider theme={theme}>
      <Button variant='primary'>Default primary</Button>
    </ThemeProvider>
  )
}

export const DefaultSecondary = () => {
  return (
    <ThemeProvider theme={theme}>
      <Button variant='secondary'>Default secondary</Button>
    </ThemeProvider>
  )
}

export const AirbnbPrimary = () => {
  return (
    <ThemeProvider theme={airbnb}>
      <Button variant='primary'>Airbnb primary</Button>
    </ThemeProvider>
  )
}

export const AirbnbSecondary = () => {
  return (
    <ThemeProvider theme={airbnb}>
      <Button variant='secondary'>Airbnb secondary</Button>
    </ThemeProvider>
  )
}

export const FigmaPrimary = () => {
  return (
    <ThemeProvider theme={figma}>
      <Button variant='primary'>Figma primary</Button>
    </ThemeProvider>
  )
}

export const FigmaSecondary = () => {
  return (
    <ThemeProvider theme={figma}>
      <Button variant='secondary'>Figma secondary</Button>
    </ThemeProvider>
  )
}

export const SpotifyPrimary = () => {
  return (
    <ThemeProvider theme={spotify}>
      <Button variant='primary'>Spotify primary</Button>
    </ThemeProvider>
  )
}

export const SpotifySecondary = () => {
  return (
    <ThemeProvider theme={spotify}>
      <Button variant='secondary'>Spotify secondary</Button>
    </ThemeProvider>
  )
}
