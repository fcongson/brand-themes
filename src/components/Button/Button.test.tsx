import { shallow } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import airbnb from '../../themes/airbnb'
import figma from '../../themes/figma'
import spotify from '../../themes/spotify'
import Button from './Button'

describe('Button', () => {
  it('Renders', () => {
    const button = shallow(<Button>Text</Button>)
    expect(button).toMatchSnapshot()
  })
  it('Renders primary', () => {
    const button = shallow(<Button variant='primary'>Text</Button>)
    expect(button).toMatchSnapshot()
  })
  it('Renders secondary', () => {
    const button = shallow(<Button variant='secondary'>Text</Button>)
    expect(button).toMatchSnapshot()
  })
  it('Renders airbnb primary', () => {
    const button = shallow(
      <ThemeProvider theme={airbnb}>
        <Button variant='primary'>Text</Button>
      </ThemeProvider>
    )
    expect(button).toMatchSnapshot()
  })
  it('Renders airbnb secondary', () => {
    const button = shallow(
      <ThemeProvider theme={airbnb}>
        <Button variant='secondary'>Text</Button>
      </ThemeProvider>
    )
  })
  it('Renders figma primary', () => {
    const button = shallow(
      <ThemeProvider theme={figma}>
        <Button variant='primary'>Text</Button>
      </ThemeProvider>
    )
    expect(button).toMatchSnapshot()
  })
  it('Renders figma secondary', () => {
    const button = shallow(
      <ThemeProvider theme={figma}>
        <Button variant='secondary'>Text</Button>
      </ThemeProvider>
    )
  })
  it('Renders spotify primary', () => {
    const button = shallow(
      <ThemeProvider theme={spotify}>
        <Button variant='primary'>Text</Button>
      </ThemeProvider>
    )
    expect(button).toMatchSnapshot()
  })
  it('Renders spotify secondary', () => {
    const button = shallow(
      <ThemeProvider theme={spotify}>
        <Button variant='secondary'>Text</Button>
      </ThemeProvider>
    )
    expect(button).toMatchSnapshot()
  })
})
