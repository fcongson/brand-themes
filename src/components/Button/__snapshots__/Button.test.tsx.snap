// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Renders 1`] = `
<button>
  Text
</button>
`;

exports[`Button Renders airbnb primary 1`] = `
<ContextProvider
  value={
    Object {
      "button": Object {
        "primary": "
  background-color: #008489;
  height: 44px;
  width: 183px;
  border-radius: 8px;
  border: 1px solid #008489;
  color: #ffffff;
  font-family: Montserrat;
  font-size: 12px;
  font-weight: 700;
  cursor: pointer;
  ",
        "secondary": "
  background-color: #ffffff;
  height: 44px;
  width: 183px;
  border-radius: 8px;
  border: 1px solid #008489;
  color: #008489;
  font-family: Montserrat;
  font-size: 12px;
  font-weight: 700;
  cursor: pointer;
  ",
      },
    }
  }
>
  <Button
    type="primary"
  >
    Text
  </Button>
</ContextProvider>
`;

exports[`Button Renders figma primary 1`] = `
<ContextProvider
  value={
    Object {
      "button": Object {
        "primary": "
  background-color: #000000;
  height: 42px;
  width: 302px;
  border-radius: 8px;
  border: 3px solid #000000;
  color: #ffffff;
  font-family: Lato;
  font-size: 16px;
  font-weight: 700;
  text-transform: none;
  cursor: pointer;
  &:hover {
    transform: translateY(-2px);
  }
  &:active {
    transform: translateY(1px);
  }
  transition: transform 200ms,background 200ms;
  ",
        "secondary": "
  background-color: #ffffff;
  height: 42px;
  width: 302px;
  border-radius: 8px;
  border: 3px solid #000000;
  color: #000000;
  font-family: Lato;
  font-size: 16px;
  font-weight: 700;
  text-transform: none;
  cursor: pointer;
  &:hover {
    transform: translateY(-2px);
  }
  &:active {
    transform: translateY(1px);
  }
  transition: transform 200ms,background 200ms;
  ",
      },
    }
  }
>
  <Button
    type="primary"
  >
    Text
  </Button>
</ContextProvider>
`;

exports[`Button Renders primary 1`] = `
<styled.button>
  Text
</styled.button>
`;

exports[`Button Renders secondary 1`] = `
<styled.button>
  Text
</styled.button>
`;

exports[`Button Renders spotify primary 1`] = `
<ContextProvider
  value={
    Object {
      "button": Object {
        "primary": "
  background-color: #1db954;
  height: 48px;
  width: 214px;
  border-radius: 500px;
  border: none;
  color: #ffffff;
  font-family: Montserrat;
  font-size: 14px;
  font-weight: 700;
  text-transform: uppercase;
  cursor: pointer;
  &:hover {
    transform: scale(1.04);
    background-color: #1ed760;
  }
  &:active {
    transform: scale(1);
    background-color: #14833b;
  }
  transition: all 33ms;
  ",
        "secondary": "
  background-color: transparent;
  height: 48px;
  width: 214px;
  border-radius: 500px;
  border: 2px solid #000000;
  color: #000000;
  font-family: Montserrat;
  font-size: 14px;
  font-weight: 700;
  text-transform: uppercase;
  cursor: pointer;
  &:hover {
    transform: scale(1.04);
  }
  &:active {
    transform: scale(1);
  }
  transition: all 33ms;
  ",
      },
    }
  }
>
  <Button
    type="primary"
  >
    Text
  </Button>
</ContextProvider>
`;

exports[`Button Renders spotify secondary 1`] = `
<ContextProvider
  value={
    Object {
      "button": Object {
        "primary": "
  background-color: #1db954;
  height: 48px;
  width: 214px;
  border-radius: 500px;
  border: none;
  color: #ffffff;
  font-family: Montserrat;
  font-size: 14px;
  font-weight: 700;
  text-transform: uppercase;
  cursor: pointer;
  &:hover {
    transform: scale(1.04);
    background-color: #1ed760;
  }
  &:active {
    transform: scale(1);
    background-color: #14833b;
  }
  transition: all 33ms;
  ",
        "secondary": "
  background-color: transparent;
  height: 48px;
  width: 214px;
  border-radius: 500px;
  border: 2px solid #000000;
  color: #000000;
  font-family: Montserrat;
  font-size: 14px;
  font-weight: 700;
  text-transform: uppercase;
  cursor: pointer;
  &:hover {
    transform: scale(1.04);
  }
  &:active {
    transform: scale(1);
  }
  transition: all 33ms;
  ",
      },
    }
  }
>
  <Button
    type="secondary"
  >
    Text
  </Button>
</ContextProvider>
`;
